### PR TITLE
[machine] create `lifecycle/` folder

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
This PR breaks up the `lifecycle.spec.ts` file and creates a `lifecycle` folder

### CODE REVIEW

File: packages/machine/test/integration/lifecycle/depositor.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+      peerB,
+      peerA,
+      ethers.utils.bigNumberify(5),
+      ethers.utils.bigNumberify(10)
+    );
+  }
+
+  /**
+   * @param amountA is the amount wallet A wants to deposit into the channel.
+   * @param amountBCumualtive is the amount wallet B already has in the channel,
```
Comment (snario) [inline packages/machine/test/integration/lifecycle/depositor.ts]: typo in a few places: "Cumulative"

File: packages/machine/test/integration/lifecycle/depositor.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    peerB: TestResponseSink,
+    threshold: ethers.utils.BigNumber
+  ) {
+    const msg = Depositor.startInstallBalanceRefundMsg(
+      peerA.signingKey.address!,
+      peerB.signingKey.address!,
+      threshold
+    );
+    const response = await peerA.runProtocol(msg);
+    expect(response.status).toEqual(cf.legacy.node.ResponseStatus.COMPLETED);
```
Comment (snario) [inline packages/machine/test/integration/lifecycle/depositor.ts]: Seems odd to me that the `ResponseStatus` type is in `cf.legacy.node`.

File: packages/machine/test/integration/lifecycle/depositor.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    // since the machine is async, we need to wait for peerB to finish up its
+    // side of the protocol before inspecting it's state
+    await cf.legacy.utils.sleep(50);
```
Comment (snario) [inline packages/machine/test/integration/lifecycle/depositor.ts]: Should take this out of `cf.legacy` and just put it in the file.

File: packages/machine/test/integration/lifecycle/depositor.ts (hunk 2 above)
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/machine/test/integration/lifecycle/depositor.ts]: `ResponseStatus` is in a few places elsewhere as well - we should fix those in a separate PR

**Pull request merged**